Libraries/Opc.Ua.Server/Session/Session.cs

```diff
@@ -936,7 +936,8 @@ internal void TraceState(string context)
             policy = null;
 
             // check for empty token.
-            if (identityToken == null || identityToken.Body == null)
+            if (identityToken == null || identityToken.Body == null ||
+                identityToken.Body.GetType() == typeof(Opc.Ua.AnonymousIdentityToken))
             {
```

Comment:
How come that this wasn't working for quite some time? Did we change something in the decoding? The aggregation server creates the session with a null user impersonation, so we reject it.

Reply:
@mregen 
It think this was not found until now because the client side (of this library) transforms the `null `user identity into `AnonymousIdentityToken` here:

https://github.com/OPCFoundation/UA-.NETStandard/blob/3548b9f195b1a5fbb8c9a815e8e8b9b87a5fe52e/Libraries/Opc.Ua.Client/Session.cs#L2205-L2209

With this I just found an ugly bug that the server was accepting AnonymousIdentity even if it was not configured for that. Your above change is fixing that!